library/core/src/intrinsics/simd.rs

```diff
@@ -0,0 +1,412 @@
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_max<T, U>(x: T) -> U;
+
+    /// Return the minimum element of a vector.
+    ///
+    /// For floating-point values, uses IEEE-754 `minNum`.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_min<T, U>(x: T) -> U;
+
+    /// Logical "and" all elements together.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_and<T, U>(x: T) -> U;
+
+    /// Logical "or" all elements together.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_or<T, U>(x: T) -> U;
+
+    /// Logical "exclusive or" all elements together.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_xor<T, U>(x: T) -> U;
+
+    /// Truncate an integer vector to a bitmask.
+    ///
+    /// Each element is truncated to a single bit and packed into the result.
+    ///
+    /// The bit order depends on the byte endianness.
+    /// The bitmask is always packed into the smallest/first bits, but the order is LSB-first for
+    /// little endian and MSB-first for big endian.
+    /// In other words, the LSB corresponds to the first vector element for little endian,
+    /// and the last vector element for big endian.
+    ///
+    /// `T` must be an integer vector.
+    ///
+    /// `U` must be either the smallest unsigned integer with at least as many bits as the length
+    /// of `T`, or the smallest array of `u8` with as many bits as the length of `T`.
+    ///
+    /// # Safety
+    /// `x` must contain only `0` and `!0`.
+    pub fn simd_bitmask<T, U>(x: T) -> U;
+
+    /// Select elements from a mask.
+    ///
+    /// For each element, if the corresponding value in `mask` is `!0`, select the element from
+    /// `if_true`.  If the corresponding value in `mask` is `0`, select the element from
+    /// `if_false`.
+    ///
+    /// `M` must be an integer vector.
+    ///
+    /// `T` must be a vector with the same number of elements as `M`.
+    ///
+    /// # Safety
+    /// `mask` must only contain `0` and `!0`.
+    pub fn simd_select<M, T>(mask: M, if_true: T, if_false: T) -> T;
+
+    /// Select elements from a bitmask.
+    ///
+    /// For each element, if the bit in `mask` is `1`, select the element from
+    /// `if_true`.  If the corresponding bit in `mask` is `0`, select the element from
+    /// `if_false`.
+    ///
+    /// The bitmask bit order matches `simd_bitmask`.
+    ///
+    /// `M` must be an unsigned integer of type matching `simd_bitmask`.
+    ///
+    /// `T` must be a vector.
+    ///
+    /// # Safety
+    /// `mask` must only contain `0` and `!0`.
```

Comment:
This makes no sense for a bitmask, does it?

However I think there is a safety requirement that the padding (the most-significant bits in the mask that are added since we are rounding up to an integer type / an integer number of `u8`) must be all-0.

Reply:
whoops, good point